src/ScanPatron/ScanPatron.js

```diff
@@ -1,7 +1,124 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash';
 
-const ScanPatron = () => (
-  <div>ScanPatron</div>
-);
+import {
+  Button,
+  Paneset,
+  Pane,
+  PaneFooter,
+} from '@folio/stripes/components';
+import { stripesConnect } from '@folio/stripes/core';
 
-export default ScanPatron;
+import ScanForm from './ScanForm';
+import PatronDetail from '../PatronDetail';
+
+const ScanPatron = ({ mutator, resources }) => {
+  const isUserProfilePicConfigEnabledForTenant = get(resources, 'userProfilePicConfig.records[0].enabled');
+  const [scannedPatronDetails, setScannedPatronDetails] = useState();
+
+  const handleScanPatron = async (barcode) => {
+    if (barcode) {
+      const query = `barcode==${barcode}`;
+      const patron = await mutator.patrons.GET({ params: { query } });
+
+      if (patron.length) {
+        setScannedPatronDetails(patron[0]);
+      } else {
+        setScannedPatronDetails();
+      }
+    }
+  };
+
+  const end = (
+    <Button
+      id="allow-access"
+      type="submit"
+      buttonStyle="primary"
+      onClick={() => {}}
+    >
+      <FormattedMessage id="ui-reading-room.allow" />
+    </Button>
+  );
+
+  const start = (
+    <>
+      <Button
+        id="cancel"
+        type="button"
+        onClick={() => {}}
+      >
+        <FormattedMessage id="ui-reading-room.cancel" />
+      </Button>
+      <Button
+        id="deny-access"
+        type="submit"
+        buttonStyle="danger"
+        onClick={() => {}}
+      >
+        <FormattedMessage id="ui-reading-room.deny" />
+      </Button>
+    </>
+  );
+
+  const footer = (
+    <PaneFooter
+      renderStart={start}
+      renderEnd={end}
+    />
+  );
+  return (
+    <Paneset>
+      <Pane
+        id="reading-room"
+        paneTitle={<FormattedMessage id="ui-reading-room.scanPatronCard" />}
+        centerContent
+        defaultWidth="fill"
+        footer={footer}
+      >
+        <ScanForm
+          onSubmit={() => {}}
```

Comment:
Please note that the above jira is a subtask of a [story ](https://folio-org.atlassian.net/browse/UIRR-4)- hence few lines may seem insignificant. They will be improved in the scope of further sub tasks.